### PR TITLE
Add MdcTagFactory to placeholder extension.

### PR DESCRIPTION
**Pull request by pstout**

### CODE REVIEW

File: spectator-ext-placeholders/build.gradle [hunk 1]
```diff
@@ -1,3 +1,8 @@
 dependencies {
   compileApi project(':spectator-api')
+
+  // The MdcTagFactoryTest requires a logging implementation that supports the MDC
+  // functionality, which the slf4j-nop (default) and slf4j-simple implementations do not.
```
Comment (brharrington) [inline spectator-ext-placeholders/build.gradle]: What is the behavior if a user is running with an implementation that doesn't support MDC?

File: spectator-ext-placeholders/src/main/java/com/netflix/spectator/placeholders/MdcTagFactory.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import com.netflix.spectator.api.BasicTag;
+import com.netflix.spectator.api.Tag;
+import com.netflix.spectator.impl.Preconditions;
+import org.slf4j.MDC;
+
+/**
+ * A TagFactory implementation that extracts information from the Sl4fj MDC data.
+ * If the MDC for the current thread has no value associated with the specified
+ * naem at the time that the createTag method is invoked, then that method will
```
Comment (brharrington) [inline spectator-ext-placeholders/src/main/java/com/netflix/spectator/placeholders/MdcTagFactory.java]: s/naem/name/

File: spectator-ext-placeholders/build.gradle (hunk 1 above)
Comment (pstout) [inline spectator-ext-placeholders/build.gradle]: You get no tags from the MdcTagFactory as the underlying `MDC.get()` call always returns null.

File: spectator-ext-placeholders/src/main/java/com/netflix/spectator/placeholders/MdcTagFactory.java (hunk 2 above)
Comment (pstout) [inline spectator-ext-placeholders/src/main/java/com/netflix/spectator/placeholders/MdcTagFactory.java]: Fixed.

**Pull request merged**